src/cli/main.ts

```diff
@@ -26,43 +26,39 @@ const main = (...args: string[]) => {
     .map((pathName) => {
       try {
         const dataFiles = fs.readdirSync(pathName);
+
         if (dataFiles) {
-          return dataFiles
+          const resultSetsByGroupId = dataFiles
             .map((dataFile) => {
               const json = fs.readFileSync(path.join(pathName, dataFile));
               return JSON.parse(json as any);
             })
-            .map(({ results }) => (results as TaskGroupResult[]).map(processResult))
-            .reduce(
-              (acc, [serverside, clientside, interactions]) => {
-                serverside.forEach((serverResult) => {
-                  const data = acc.server[serverResult.taskName];
-                  // eslint-disable-next-line no-unused-expressions
-                  data
-                    ? data.push(serverResult.value)
-                    : (acc.server[serverResult.taskName] = [serverResult.value]);
-                });
-
-                clientside.forEach((clientResult) => {
-                  const data = acc.client[clientResult.taskName];
-                  // eslint-disable-next-line no-unused-expressions
-                  data
-                    ? data.push(clientResult.value)
-                    : (acc.client[clientResult.taskName] = [clientResult.value]);
+            .map(({ results }) => results as TaskGroupResult[])
+            .reduce((resultSets, taskGroupResults) => {
+              const resultsByGroupId = taskGroupResults.reduce(
+                (resultsByGroupId, { groupId, map }: TaskGroupResult) => ({
+                  ...resultsByGroupId,
+                  [groupId]: convertToTaskValueMap(map),
+                }),
+                {} as ResultByGroupId,
+              );
+
+              Object.entries(resultsByGroupId).forEach(([groupId, taskValueMap]) => {
```

Comment:
nit: wondering if we can re-arrange some of these so the it's a bit more pure functional than all these forEach's with side effects. Can we do it cleanly or too tricky? Also I know I started this but always looking for ways to make it nicer.